compat/media/camera_service.cpp

```diff
@@ -23,7 +23,7 @@
 #include "media_recorder_factory.h"
 #include "media_recorder.h"
 
-#include <media/camera_record_service.h>
+//#include <media/camera_record_service.h>
```

Comment:
I would make this include conditional with #if ANDROID_VERSION_MAJOR==5

Reply:
That would only work for Halium or other projects using Ubuntu based Android repos.

Reply:
Is Halium/UBports the only consumer of the compat portions?

Reply:
@fredl

> Is Halium/UBports the only consumer of the compat portions?

Some compat layers are also used by SailfishOS but this one isn't.

Reply:
Should we remove it instead of commenting out since the header isn't in the tree anymore?

Reply:
> pmOS is using mainline kernels.

pmOS does not limit device ports to specific kernel, some ports are using downstream android kernels and Halium, too.

Reply:
@minlexx In that case a build flag would make sense to me to allow for flexibility by the different OS-es using libhybris in their own way.

Reply:
a build time flag would make sense to me, in theory you could even check for the existence of the header(s) in configure.ac like we did with HAS_HWCOMPOSER2_HEADERS and similar ones.

Reply:
How does it look now? The condition can be fulfilled both via configure and from the Android side using `CAMERA_SERVICE_WANT_UBUNTU_HEADERS := 1`.